### ISSUE TITLE
only show user's projects in uploader project selector

### ISSUE THREAD

**Issue by kueda** [inaturalist/inaturalist#1025]
Most people haven't joined a million projects, so having the ones they've joined pop up as the default auto-complete options (instead of all projects) would be pretty useful. Sorted by the number of obs they've added to those projects might be even better, but not a requirement. I also don't think this absolutely has to be done before making the uploader our primary way to add obs, but definitely a nice-to-have.

**Comment by pleary** [comment]
I'm a little unclear about the wording here, and what people are asking for in their comments. This says "default auto-complete options", but right now when you click on the project autocomplete, the "default" options that pop up are ones you've joined, the same selections as the project menu in the header - it's just not all of them because there could be hundreds. When you search, you're searching everything.

In the bulk editor, the projects list showed there is ALL the projects a user has joined in a scrolling div. So what is being requested here? One of more of the following?:

1. Make it so the default list that pops up when the empty project autocomplete field is clicked is a users COMPLETE list of projects joined. I'd presumably need to make the autocomplete list there a scrolling div, which isn't something we do elsewhere with autocompletes
2. Make it so when a user searches in the project autocomplete, the only options returned are projects the user has joined. i.e. users cannot search for projects they have not joined.

I *think* more people are wanting 1, and I worry that 2 could be confusing since you don't technically need to join projects to contribute to them. If we do 1, should we just sort them alphabetically like the batch edit form? Someone requested recent ones at the top, but I think that'll make it harder to scan larger lists

**Comment by kueda** [comment]
Good points, sorry for not being more clear. 1 would probably meet people's needs, not 2. Taking a quick look at the data from the people who have been complaining, it seems most people who manually add things to projects have like 1-5 that they add to a lot, and maybe 10-50 that they add to sometimes. The distributions are actually a bit more flat than I would have expected, was thinking more like 1-3 regular projects, ~10 sometimes projects.

Sort I'm not sure about. I feel like when we ask the user to scroll, the tool has become less efficient than typing in a name, so alphabetizing would only really help if the user has only joined a handful of projects, but if they hate typing, love scrolling, and have a lot of projects, then alphabetizing is essential.

I think I'm over-thinking this. I'd just go with your option 1.

**Comment by pleary** [comment]
Option 1 is on Staging for testing
